File: scala_example/WordCount.scala
import org.apache.spark.{SparkConf, SparkContext}

class SparkWordCount {
  def wdCount(fileName: String): Unit = {

    //Create a SparkContext to initialize Spark
    val conf = new SparkConf()
    conf.setMaster("local")
    conf.setAppName("Word Count")
    val sc = new SparkContext(conf)

    // Load the text into a Spark RDD, which is a distributed representation of each line of text
    //val textFile = sc.textFile("/home/qzhang/Downloads/input")
    val textFile = sc.textFile(fileName)

    //word count
    val counts = textFile.flatMap(line => line.split(" "))
      .map(word => (word, 1))
      .reduceByKey(_ + _)

    counts.foreach(println)
    System.out.println("Total words: " + counts.count());
    // counts.saveAsTextFile("/home/qzhang/Downloads/writeback");
  }
}


object WordCount {
//    def main(args: Array[String]): Unit = {
//      println("HelloWorld!")
//    }
  def main(args: Array[String]): Unit = {
     val obj = new SparkWordCount()
     obj.wdCount("/home/qzhang/Downloads/input");
  }
}

File: customarray/src/edu/ucla/cs/bigfuzz/sparkprogram/WordCount.java
package edu.ucla.cs.bigfuzz.sparkprogram;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function2;
import scala.Tuple2;

import java.io.Serializable;
import java.util.Arrays;
import java.util.List;

public class WordCount implements Serializable {

    public static void main(String[] args) {
        SparkConf sparkConf = new SparkConf()
                .setAppName("Example Spark App")
                .setMaster("local[*]"); // Delete this line when submitting to a cluster
        JavaSparkContext sparkContext = new JavaSparkContext(sparkConf);

//        JavaRDD<String> lines = sparkContext.textFile(inputFile);
        JavaRDD<String> lines = sparkContext.textFile("dataset/data/");

        JavaRDD words = lines.flatMap(s -> Arrays.asList(s.split(" ")).iterator());

        JavaPairRDD counts = words.mapToPair(s->new Tuple2<>(s, 1)).reduceByKey(new Function2<Integer, Integer, Integer>() {
            @Override
            public Integer call(Integer first, Integer second) throws Exception {
                return first + second;
            }
        });

        List<Tuple2<String, Integer>> output = counts.collect();
        for (Tuple2<?,?> tuple : output) {
            System.out.println(tuple._1() + ": " + tuple._2());
        }

//        counts.saveAsTextFile("/home/qzhang/Downloads/writeback");
    }
}
